### PR TITLE
Put batchnorm in eval mode in validation phase

### LINKED ISSUE #44: `model.train()` usage
The code uses `model.train()` during both train and valid phases, wouldn't this effect the running stats for batchnorm layers. Is this intentional?

https://github.com/princeton-vl/pytorch_stacked_hourglass/blob/ed91059e874f35089dd3a8e692fa895929785b91/task/pose.py#L107-L132

### PR DESCRIPTION
**Pull request by ashutoshsingh0223**
- Only put batchnorm layers in eval mode when in validation phase.
   - Can't put the whole `Trainer`  in eval mode since loss calculation requires it to be in train mode.
   - Editing `Trainer`  class itself might be an overkill to fix this issue. 
- No changes in other parts of the pipeline.
- Closes #44

### CODE REVIEW

**Comment by ashutoshsingh0223** [comment]
I will post the results soon.

**Comment by crockwell** [comment]
thanks!

**Comment by ashutoshsingh0223** [comment]
These are the results from epoch 150. The model used is the default model defined in the code(i.e. I did not change anything except for putting batchnorm layers in eval mode for passes with validation data.)

Command used for training - 
`python train.py -e test_run_001`


```
=> loading checkpoint 'exp/test_run_001'
=> loaded checkpoint 'exp/test_run_001' (epoch 150)
all :
Val PCK @, 0.5 , total : 0.901 , count: 44239
Tra PCK @, 0.5 , total : 0.958 , count: 4443
Val PCK @, 0.5 , ankle : 0.803 , count: 4234
Tra PCK @, 0.5 , ankle : 0.893 , count: 392
Val PCK @, 0.5 , knee : 0.849 , count: 4963
Tra PCK @, 0.5 , knee : 0.926 , count: 499
Val PCK @, 0.5 , hip : 0.891 , count: 5777
Tra PCK @, 0.5 , hip : 0.945 , count: 587
Val PCK @, 0.5 , pelvis : 0.926 , count: 2878
Tra PCK @, 0.5 , pelvis : 0.973 , count: 293
Val PCK @, 0.5 , thorax : 0.981 , count: 2932
Tra PCK @, 0.5 , thorax : 0.997 , count: 298
Val PCK @, 0.5 , neck : 0.979 , count: 2932
Tra PCK @, 0.5 , neck : 0.993 , count: 298
Val PCK @, 0.5 , head : 0.96 , count: 2931
Tra PCK @, 0.5 , head : 0.993 , count: 297
Val PCK @, 0.5 , wrist : 0.85 , count: 5837
Tra PCK @, 0.5 , wrist : 0.939 , count: 589
Val PCK @, 0.5 , elbow : 0.9 , count: 5867
Tra PCK @, 0.5 , elbow : 0.963 , count: 592
Val PCK @, 0.5 , shoulder : 0.955 , count: 5888
Tra PCK @, 0.5 , shoulder : 0.992 , count: 598


visible :
Val PCK @, 0.5 , total : 0.942 , count: 33996
Tra PCK @, 0.5 , total : 0.972 , count: 3879
Val PCK @, 0.5 , ankle : 0.888 , count: 3177
Tra PCK @, 0.5 , ankle : 0.938 , count: 340
Val PCK @, 0.5 , knee : 0.909 , count: 3854
Tra PCK @, 0.5 , knee : 0.949 , count: 448
Val PCK @, 0.5 , hip : 0.917 , count: 4206
Tra PCK @, 0.5 , hip : 0.946 , count: 503
Val PCK @, 0.5 , pelvis : 0.954 , count: 1620
Tra PCK @, 0.5 , pelvis : 0.977 , count: 221
Val PCK @, 0.5 , thorax : 0.992 , count: 1691
Tra PCK @, 0.5 , thorax : 1.0 , count: 236
Val PCK @, 0.5 , neck : 0.979 , count: 2932
Tra PCK @, 0.5 , neck : 0.993 , count: 298
Val PCK @, 0.5 , head : 0.96 , count: 2931
Tra PCK @, 0.5 , head : 0.993 , count: 297
Val PCK @, 0.5 , wrist : 0.934 , count: 4395
Tra PCK @, 0.5 , wrist : 0.973 , count: 490
Val PCK @, 0.5 , elbow : 0.95 , count: 4590
Tra PCK @, 0.5 , elbow : 0.977 , count: 511
Val PCK @, 0.5 , shoulder : 0.972 , count: 4600
Tra PCK @, 0.5 , shoulder : 0.993 , count: 535


not visible :
Val PCK @, 0.5 , total : 0.765 , count: 10243
Tra PCK @, 0.5 , total : 0.862 , count: 564
Val PCK @, 0.5 , ankle : 0.55 , count: 1057
Tra PCK @, 0.5 , ankle : 0.596 , count: 52
Val PCK @, 0.5 , knee : 0.642 , count: 1109
Tra PCK @, 0.5 , knee : 0.725 , count: 51
Val PCK @, 0.5 , hip : 0.82 , count: 1571
Tra PCK @, 0.5 , hip : 0.94 , count: 84
Val PCK @, 0.5 , pelvis : 0.891 , count: 1258
Tra PCK @, 0.5 , pelvis : 0.958 , count: 72
Val PCK @, 0.5 , thorax : 0.966 , count: 1241
Tra PCK @, 0.5 , thorax : 0.984 , count: 62
Val PCK @, 0.5 , neck : 0.0 , count: 0
Tra PCK @, 0.5 , neck : 0.0 , count: 0
Val PCK @, 0.5 , head : 0.0 , count: 0
Tra PCK @, 0.5 , head : 0.0 , count: 0
Val PCK @, 0.5 , wrist : 0.596 , count: 1442
Tra PCK @, 0.5 , wrist : 0.768 , count: 99
Val PCK @, 0.5 , elbow : 0.721 , count: 1277
Tra PCK @, 0.5 , elbow : 0.877 , count: 81
Val PCK @, 0.5 , shoulder : 0.892 , count: 1288
```

**Comment by crockwell** [comment]
Thanks so much Ashutosh!
